src/Network/Broadcast/OutboundQueue.hs

```diff
@@ -232,12 +253,63 @@ defaultEnqueuePolicy NodeEdge = go
         -- not forwarded
       ]
     go MsgRequestBlockHeaders = [
-        EnqueueAll NodeRelay (MaxAhead 20) PLowest
+        EnqueueAll NodeRelay (MaxAhead 1) PHigh
+      ]
+    go (MsgRequestBlock _) = [
+        -- Edge nodes can only talk to relay nodes
+        EnqueueOne [NodeRelay] UnlimitedMaxAhead PHigh
+      ]
+    go (MsgMPC _) = [
+        -- not relevant
+      ]
+
+-- | Default enqueue policy for exchange nodes
+defaultEnqueuePolicyEdgeExchange :: EnqueuePolicy nid
+defaultEnqueuePolicyEdgeExchange = go
+  where
+    -- Enqueue policy for edge nodes
+    go :: EnqueuePolicy nid
+    go (MsgTransaction OriginSender) = [
+        EnqueueAll NodeRelay (MaxAhead 6) PLow
+      ]
+    go (MsgTransaction (OriginForward _)) = [
+        -- don't forward transactions that weren't created at this node
+      ]
+    go (MsgAnnounceBlockHeader _) = [
+        -- not forwarded
+      ]
+    go MsgRequestBlockHeaders = [
+        EnqueueAll NodeRelay (MaxAhead 1) PHigh
+      ]
+    go (MsgRequestBlock _) = [
+        -- Edge nodes can only talk to relay nodes
+        EnqueueOne [NodeRelay] UnlimitedMaxAhead PHigh
+      ]
+    go (MsgMPC _) = [
+        -- not relevant
+      ]
+
+-- | Default enqueue policy for edge nodes using P2P
+defaultEnqueuePolicyEdgeP2P :: EnqueuePolicy nid
+defaultEnqueuePolicyEdgeP2P = go
+  where
+    -- Enqueue policy for edge nodes
+    go :: EnqueuePolicy nid
+    go (MsgTransaction OriginSender) = [
+        EnqueueAll NodeRelay (MaxAhead 3) PLow
+      ]
+    go (MsgTransaction (OriginForward _)) = [
+        -- don't forward transactions that weren't created at this node
+      ]
```

Comment:
Actually for P2P nodes we'll do exactly that, so just use  `MsgTransaction _` rather than `MsgTransaction OriginSender` in the first case above and omit the second case.